UnityProject/Assets/Scripts/Player/PlayerSprites.cs

```diff
@@ -110,52 +142,119 @@ private IEnumerator WaitForLoad()
-			FaceDirection( currentDirection );
+			LocalFaceDirection( currentBodyDirection );
 		}
 		while(string.IsNullOrEmpty(characterData)){
 			yield return YieldHelper.DeciSecond;
 		}
-		FaceDirectionSync(currentDirection);
+		FaceBodyDirectionSync(currentBodyDirection);
+		FaceGhostDirectionSync(currentGhostDirection);
 		if (PlayerManager.LocalPlayer != gameObject)
 		{
 			UpdateCharacterSprites(characterData);
 		}
 	}
 
-	public void AdjustSpriteOrders(int offsetOrder)
+	private void OnRotationStart(RotationOffset fromCurrent, bool isInitialRotation)
+	{
+		//ignore the initial rotation message because we determine initial rotation from the
+		//currentBodyDirection syncvar in playerSprites
+		if (!isInitialRotation)
+		{
+			//determine our destination rotation
+			if (playerMove.isGhost)
+			{
+				destinationOrientation = currentGhostDirection.Rotate(fromCurrent);
+			}
+			else
+			{
+				destinationOrientation = currentBodyDirection.Rotate(fromCurrent);
+			}
+
+			isMatrixRotating = true;
+		}
+	}
+
+	private void OnRotationEnd(RotationOffset fromCurrent, bool isInitialRotation)
 	{
-		foreach (SpriteRenderer s in GetComponentsInChildren<SpriteRenderer>())
+
+		//ignore the initial rotation message because we determine initial rotation from the
+		//currentBodyDirection syncvar in playerSprites
+		if (!isInitialRotation)
 		{
-			int newOrder = s.sortingOrder;
-			newOrder += offsetOrder;
-			s.sortingOrder = newOrder;
+			LocalFaceDirection(destinationOrientation);
+			isMatrixRotating = false;
 		}
+
 	}
 
 	/// <summary>
-	/// Change current facing direction to match direction
+	/// Change current facing direction to match direction (it's a Command so it's invoked on the server by
+	/// the server itself or the client)
 	/// </summary>
 	/// <param name="direction">new direction</param>
 	[Command]
-	public void CmdChangeDirection(Orientation direction)
+	private void CmdChangeDirection(Orientation direction)
 	{
-		FaceDirection(direction);
+		LocalFaceDirection(direction);
 	}
 
-	//turning character input and sprite update for local only! (prediction)
-	public void FaceDirection(Orientation direction)
+	/// <summary>
+	/// Locally changes the direction of this player to face the specified direction but doesn't tell the server.
+	/// If this is a client, only changes the direction locally and doesn't inform other players / server.
+	/// If this is on the server, the direction change will be sent to all clients due to the syncvar.
+	///
+	/// Changes ghost direction if they are a ghost, otherwise changes body direction.
+	///
+	/// Does nothing if player is down
+	/// </summary>
+	/// <param name="direction"></param>
+	public void LocalFaceDirection(Orientation direction)
 	{
-		if(playerSync.isBumping)
+		if (registerPlayer.IsDown || playerSync.isBumping)
 		{
 			//Don't face while bumping is occuring on this frame
+			//or when player is down
 			return;
 		}
-		SetDir(direction);
+		if (playerMove.isGhost)
```

Comment:
If the player is down this condition is never reached therefore disabling ghost directional sprites from being updated.

Reply:
going to push through a fix for this after merging so we can get the PR through